src/mqtt/MQTT.cpp

```diff
@@ -43,9 +43,20 @@ void MQTT::onPublish(char *topic, byte *payload, unsigned int length)
         JSONValue *json_value = JSON::Parse(payloadStr);
         if (json_value != NULL) {
             LOG_INFO("JSON Received on MQTT, parsing..\n");
+
             // check if it is a valid envelope
             JSONObject json;
             json = json_value->AsObject();
+
+            // parse the channel name from the topic string
+            char *ptr = strtok(topic, "/");
+            for (int i = 0; i < 3; i++) {
```

Comment:
strtok alters the topic string by replacing the / with \0 but it's not used after the operation. I hope it will be reseeded by the next call to onPublish.